okhttp-logging-interceptor/src/test/java/okhttp3/logging/HttpLoggingInterceptorTest.java

```diff
@@ -641,6 +642,26 @@ private void bodyGetNoBody(int code) throws IOException {
         .assertNoMoreLogs();
   }
 
+    @Test public void connectFail() throws IOException {
+        setLevel(Level.BASIC);
+        url = new HttpUrl.Builder()
+            .scheme("http")
+            .host("okhttp-notfound.com")
```

Comment:
How about a custom `Dns` instead. Tests should absolutely not be connecting to real things (even real DNS servers).